test-requirements.txt

```diff
@@ -1,16 +1,85 @@
--r mypy-requirements.txt
--r build-requirements.txt
-attrs>=18.0
-black==23.9.1  # must match version in .pre-commit-config.yaml
-filelock>=3.3.0
-# lxml 4.9.3 switched to manylinux_2_28, the wheel builder still uses manylinux2014
-lxml>=4.9.1,<4.9.3; (python_version<'3.11' or sys_platform!='win32') and python_version<'3.12'
-pre-commit
+#
+# This file is autogenerated by pip-compile with Python 3.11
+# by the following command:
+#
+#    pip-compile --output-file=test-requirements.txt --unsafe-package=typed-ast test-requirements.in
```

Comment:
Why do we need `--unsafe-package=typed-ast` part?

Reply:
I mention this in PR description, it's completely arbitrary, passing this results in pip-compile locking setuptools which is desirable since setuptools has caused us CI failures in the past